cli/azd/pkg/infra/provisioning/manager.go

```diff
@@ -1,72 +1,104 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
 package provisioning
 
 import (
 	"context"
 	"fmt"
+	"io"
+	"log"
 	"strings"
 
+	"github.com/azure/azure-dev/cli/azd/pkg/azureutil"
 	"github.com/azure/azure-dev/cli/azd/pkg/environment"
 	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/azure/azure-dev/cli/azd/pkg/output"
 	"github.com/azure/azure-dev/cli/azd/pkg/spin"
 	"github.com/azure/azure-dev/cli/azd/pkg/tools"
 	"github.com/azure/azure-dev/cli/azd/pkg/tools/azcli"
-	"github.com/azure/azure-dev/cli/azd/pkg/tools/bicep"
 )
 
+// Manages the orchestration of infrastructure provisioning
 type Manager struct {
 	azCli       azcli.AzCli
 	env         environment.Environment
 	provider    Provider
-	interactive bool
+	formatter   output.Formatter
+	writer      io.Writer
 	console     input.Console
+	interactive bool
 }
 
 // Prepares for an infrastructure provision operation
-func (m *Manager) Preview(ctx context.Context, interactive bool) (*PreviewResult, error) {
-	previewResult, err := m.preview(ctx, interactive)
+func (m *Manager) Preview(ctx context.Context) (*PreviewResult, error) {
+	previewResult, err := m.preview(ctx)
 	if err != nil {
 		return nil, err
 	}
 
-	updated, err := m.ensureParameters(ctx, &previewResult.Preview)
-	if err != nil {
-		return nil, err
-	}
+	return previewResult, nil
+}
 
-	if updated {
-		if err := m.provider.UpdatePlan(ctx, previewResult.Preview); err != nil {
-			return nil, fmt.Errorf("updating deployment parameters: %w", err)
-		}
+// Gets the latest deployment details for the specified scope
+func (m *Manager) GetDeployment(ctx context.Context, scope Scope) (*DeployResult, error) {
+	var deployResult *DeployResult
+
+	err := m.runAction("Retrieving Azure Deployment", m.interactive, func(spinner *spin.Spinner) error {
+		queryTask := m.provider.GetDeployment(ctx, scope)
 
-		if err := m.env.Save(); err != nil {
-			return nil, fmt.Errorf("saving env file: %w", err)
+		go func() {
+			for progress := range queryTask.Progress() {
+				m.updateSpinnerTitle(spinner, progress.Message)
+			}
+		}()
+
+		go m.monitorInteraction(spinner, queryTask.Interactive())
+
+		result, err := queryTask.Await()
+		if err != nil {
+			return err
 		}
+
+		deployResult = result
+
+		return nil
+	})
+
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving deployment: %w", err)
 	}
 
-	return previewResult, nil
+	m.console.Message(ctx, "Azure env refresh completed 👍")
```

Comment:
Surprised this ends up writing output, I can imagine cases where we call `GetDeployment` outside the context of a refresh.